api/test/robot_tests/plugin_manager.robot

```diff
@@ -2,20 +2,215 @@
+
+DELETE /plugin-manager/{id} with invalid id
+    ${response}=    DELETE    ${API_URL}/plugin-manager/3829029    expected_status=${STATUS_INTERNAL_SERVER_ERROR}
+    Should Be Equal As Strings
+    ...    ${response.json()['message']}
+    ...    deleting plugin 3829029: no plugin matching id 3829029
+
+POST /plugin-manager/{id}/execute with invalid body
+    ${data}=    Create Dictionary    command=test
+    ${response}=    POST
+    ...    ${API_URL}/plugin-manager/invalid/execute
+    ...    expected_status=${STATUS_UNPROCESSABLE_ENTITY}
+    ...    json=${data}
+    Should Be Equal As Strings    ${response.json()['code']}    606
+    Should Be Equal As Strings
+    ...    ${response.json()['message']}
+    ...    body.command in body should be one of [update stop start restart]
+
+POST /plugin-manager/register with invalid id
+    ${data}=    Create Dictionary
+    ...    id=1
+    ...    name=ut aliqua non
+    ...    author=adipisicing
+    ...    description=minim consectetur dolore,
+    ...    logo=id et sunt irure,
+    ...    home=sunt
+    ...    api_spec=culpa enim sint aliqua
+    ...    url=oluptate
+    ${response}=    POST
+    ...    ${API_URL}/plugin-manager/register
+    ...    expected_status=${STATUS_NOT_FOUND}
+    ...    json=${data}
+
+    Should Be Equal As Strings    ${response.json()['code']}    Plugin-0001
+    Should Be Equal As Strings    ${response.json()['message']}    get plugin error: no plugin matching id 1
+
+POST /plugin-manager/register with invalid body
+    ${data}=    Create Dictionary
+    ...    id=-65217
+    ...    name=ut aliqua non
+    ...    author=adipisicing
+    ...    description=minim consectetur dolore,
+    ...    logo=id et sunt irure,
+    ...    home=sunt
+    ...    api_spec=culpa enim sint aliqua
+    ${response}=    POST
+    ...    ${API_URL}/plugin-manager/register
+    ...    expected_status=${STATUS_UNPROCESSABLE_ENTITY}
+    ...    json=${data}
+    Should Be Equal As Strings    ${response.json()['code']}    602
+    Should Be Equal As Strings    ${response.json()['message']}    body.url in body is required
+
+POST /plugin-manager/{id}/execute with NotImplemented update command
+    ${id}=    Get Plugin ID From Author and Name    massalabs    hello world
+    ${data}=    Create Dictionary    command=update
+    ${response}=    POST
+    ...    ${API_URL}/plugin-manager/${id}/execute
+    ...    expected_status=${STATUS_NOT_IMPLEMENTED}
+    ...    json=${data}
+
+
+*** Keywords ***
+Suite Setup
+    [Documentation]    Suite Setup specific to this test suite
+    Basic Suite Setup
+    Delete all plugins
+
+Delete all plugins
+    Log To Console    Deleting all plugins
+    ${response}=    GET    ${API_URL}/plugin-manager
+    FOR    ${element}    IN    @{response.json()}
+        ${response}=    DELETE    ${API_URL}/plugin-manager/${element['id']}
+        Status Should Be    ${STATUS_NO_CONTENT}
+    END
```

Comment:
This is dangerous: you will delete all the plugins. Even thoses that you didn't install. Is it on purpose?

Reply:
It is on purpose. I was thinking we would need multiple plugins to test the plugin manager properly, but only one seems to be enough.

Since we only use the `hello-world` plugin for now, we could just delete it and reinstall it.

WDYT?

Reply:
Ok,but should have a big warning in contributing.

Reply:
I already made one [in the API tests README](https://github.com/massalabs/thyra/blob/1da9a7385d2508a31d65e5b4da528fc4380b9030/api/test/README.md?plain=1#L23)

I will modify the test suite to only delete the `hello-world` plugin tho. It will be a life improvement for devs since they wont need to reinstall all their plugins after running tests.